test/unit/StakingPool/withdraw.js

```diff
@@ -0,0 +1,573 @@
+    await increaseTime(TRANCHE_DURATION);
+
+    const userBalanceBefore = await nxm.balanceOf(user.address);
+    const randomUserBalanceBefore = await nxm.balanceOf(randomUser.address);
+
+    expect(
+      await stakingPool
+        .connect(randomUser)
+        .withdraw([{ tokenId: depositNftId, withdrawStake, withdrawRewards, trancheIds }]),
+    ).to.not.be.reverted;
+
+    const { stake } = await calculateStakeAndRewardsWithdrawAmounts(stakingPool, deposit, firstActiveTrancheId);
+
+    const userBalanceAfter = await nxm.balanceOf(user.address);
+    const randomUserBalanceAfter = await nxm.balanceOf(randomUser.address);
+
+    expect(randomUserBalanceAfter).to.eq(randomUserBalanceBefore);
+    expect(userBalanceAfter).to.eq(userBalanceBefore.add(stake));
+  });
+
+  it('should emit some event', async function () {
+    const {
+      cover,
+      stakingPool,
+      accounts: {
+        members: [user],
+      },
+    } = this;
+
+    const { amount, tokenId, destination, depositNftId } = depositToFixture;
+    const coverSigner = await ethers.getImpersonatedSigner(cover.address);
+
+    const TRANCHES_NUMBER = 3;
+    const trancheIds = [];
+
+    const withdrawStake = true;
+    const withdrawRewards = true;
+
+    for (let i = 0; i < TRANCHES_NUMBER; i++) {
+      const { firstActiveTrancheId: currentTranche } = await getTranches();
+
+      await stakingPool.connect(user).depositTo([
+        {
+          amount,
+          trancheId: currentTranche,
+          tokenId,
+          destination,
+        },
+      ]);
+
+      trancheIds.push(currentTranche);
+      await stakingPool.connect(coverSigner).allocateCapacity(allocationRequest, allocationConfig);
+      await increaseTime(TRANCHE_DURATION);
+      await mineNextBlock();
+    }
+
+    const depositsBeforeWithdraw = {};
+    for (const tranche of trancheIds) {
+      depositsBeforeWithdraw[tranche] = await stakingPool.deposits(depositNftId, tranche);
+    }
+
+    await stakingPool.connect(user).withdraw([{ tokenId: depositNftId, withdrawStake, withdrawRewards, trancheIds }]);
+
+    const stakes = [];
+    const rewards = [];
+    for (const tranche of trancheIds) {
+      const { rewards: currentReward, stake } = await calculateStakeAndRewardsWithdrawAmounts(
+        stakingPool,
+        depositsBeforeWithdraw[tranche],
+        tranche,
+      );
+
+      stakes.push(stake);
+      rewards.push(currentReward);
+    }
+
+    await expect(
+      stakingPool.connect(user).withdraw([{ tokenId: depositNftId, withdrawStake, withdrawRewards, trancheIds }]),
+    )
+      .to.emit(stakingPool, 'Withdraw')
+      .withArgs(user.address, depositNftId, trancheIds[1], stakes[1], rewards[1])
+      .to.emit(stakingPool, 'Withdraw')
+      .withArgs(user.address, depositNftId, trancheIds[2], stakes[2], rewards[2])
+      .to.emit(stakingPool, 'Withdraw')
+      .withArgs(user.address, depositNftId, trancheIds[2], stakes[2], rewards[2]);
```

Comment:
This should have index 0, I'm still debugging why is not the same as the emitted args.